include/envoy/http/async_client.h

```diff
@@ -50,6 +83,39 @@ class AsyncClient {
     virtual void cancel() PURE;
   };
 
+  /**
+   * An in-flight HTTP stream
+   */
+  class Stream {
+  public:
+    virtual ~Stream() {}
+
+    /***
+     * Send headers to the stream
+     * @param headers supplies the headers to send
+     * @param end_stream supplies whether this is a header only request
+     */
+    virtual void sendHeaders(HeaderMap& headers, bool end_stream) PURE;
+
+    /***
+     * Send data to the stream
+     * @param data supplies the data to send
+     * @param end_stream supplies whether this is the last data
+     */
+    virtual void sendData(Buffer::Instance& data, bool end_stream) PURE;
+
+    /***
+     * Send trailers. This implicitly ends the stream.
+     * @param trailers supplies the trailers to send
+     */
+    virtual void sendTrailers(HeaderMap& trailers) PURE;
+
+    /***
+     * Close the stream.
+     */
+    virtual void close() PURE;
```

Comment:
Sorry you are right, I just looked at the code again. The fact that there is no reason is due to how we are wrapping the router filter to do all of this. I would just rename to reset() for clarity since IMO that makes more sense in the streaming case.